sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/routing/LocationCache.java

```diff
@@ -332,37 +333,83 @@ public boolean shouldRefreshEndpoints(Utils.ValueHolder<Boolean> canRefreshInBac
         DatabaseAccountLocationsInfo currentLocationInfo = this.locationInfo;
         String mostPreferredLocation = Utils.firstOrDefault(currentLocationInfo.preferredLocations);
 
+        String mostPreferredEffectiveReadLocation = StringUtils.EMPTY;
```

Comment:
what if every time we refresh the location we also update the preferredLocations to be effectivePreferredLocations, then we might do not need to specialize the handling when connectionPolicy.preferredRegions being empty situation in this method and the following methods

Reply:
Will address it shortly.

Reply:
or maybe we need to maintain preferredRegions, but add another one effectivePreferredRegions in the locationInfo, depending on how the preferredRegions is used currently

Reply:
Yeah - the `DatabaseAccount` / `LocationCache` refresh part is greatly simplified now.